racket/collects/racket/match/compiler.rkt

```diff
@@ -551,5 +543,27 @@
       (with-syntax ([(fns ... [_ (lambda () body)]) fns])
         (let/wrap #'(fns ...) #'body)))]))
 
+;; find-seen : Id (Listof (Pairof Id (U #f Id))) -> (U #f Id)
+(define (find-seen v seen)
+  (cond
+    [(for/or ([e (in-list seen)])
+       (let ([v* (car e)] [id (cdr e)])
+         (and (bound-identifier=? v v*) (or id (list v v*)))))
+     =>
+     (lambda (id)
+       (if (identifier? id)
+           id
+           (begin
+             (log-error "non-linear pattern used in `match` with ... at ~a and ~a"
```

Comment:
If this is supposed to work now this log should be removed.

Reply:
I didn't want things like this to change for now:
```racket
(match '(1 2 3 4)
  [(list a ... a) a])
```
Order matters. When the `a` without ellipses comes first, that's when it works as a nonlinear pattern:
```racket
(match '(1 1 1 1)
  [(list a a ...) a])
```

Reply:
I've updated the PR, it now makes nonlinear pattern declarations syntax errors if the first use is in the scope of a `...` pattern that doesn't encompass all uses.